src/client/clientmap.cpp

```diff
@@ -630,23 +652,27 @@ void ClientMap::touchMapBlocks()
 		*/
 
 		for (MapBlock *block : sectorblocks) {
-			/*
-				Compare block position to camera position, skip
-				if not seen on display
-			*/
+			MapBlockMesh *mesh = block->mesh;
 
-			if (!block->mesh) {
```

Comment:
This, and following changes are unrelated to bringing loops back.
Did you do a brief perf test on this? Might be quite more expensive.

Also, even with 1x1x1 mesh_chunks there are blocks without a mesh, I am not sure we want to reset the usage counters for those. And I mean I am really not sure one way or the other.

Reply:
I've noticed that with small values for client_unload_unused_data_timeout the wield mesh flickers when flying about ground without this patch. This is because the air block the player is in gets unloaded and replaced with 'ignore' blocks which do not store light. Keeping air blocks loaded in the client prevents this unloading cycle and makes the picture stable even when client is configured to unload blocks aggressively.